Packs/FeedGithub/Integrations/FeedGithub/FeedGitHub.py

```diff
@@ -0,0 +1,514 @@
+        response = self._http_request("GET", full_url=f"{self._base_url}commits", resp_type="response")
+        while "next" in response.links:
+            data = response.json()
+            all_commits.extend(data)
+            response = self._http_request("GET", full_url=response.links["next"]["url"], resp_type="response")
+        return all_commits[-1]["sha"]
+
+    def get_files_between_commits(self, base: str, head: str) -> list:
+        """
+        Retrieves the list of files modified between two commits.
+
+        Args:
+            base (str): The SHA of the base commit.
+            head (str): The SHA of the head commit.
+
+        Returns:
+            list: A list of files modified between the specified base and head commits.
+        """
+        return self._http_request("GET", f"compare/{base}...{head}")["files"]
+
+
+def parsing_files_by_status(commits_files: list) -> list:
+    """
+    Parses files from a list of commit files based on their status.
+
+    Args:
+        commits_files (list): A list of dictionaries representing commit files.
+
+    Returns:
+        list: A list of URLs for files that are added or modified.
+    """
+    relevant_files: list[dict] = []
+    for file in commits_files:
+        if file.get("status") == "added" or file.get("status") == "modified":
+            relevant_files.append(file.get("raw_url"))
+    return relevant_files
+
+
+def get_files_token(client: Client, relevant_files: list[str]):
+    """
+    Retrieves content of relevant files based on specified extensions.
+
+    Args:
+        client (Client): An instance of the client used for HTTP requests.
+        relevant_files (list): A list of URLs for relevant files.
+
+    Returns:
+        list: A list of file contents fetched via HTTP requests.
+    """
+    extensions_to_fetch = argToList(demisto.params().get("extensions_to_fetch") or [])
+    relevant_files = [file for file in relevant_files if any(file.endswith(ext) for ext in extensions_to_fetch)]
+    return [client._http_request("GET", full_url=file, resp_type="text") for file in relevant_files]
+
+
+def get_commits_files(client: Client, last_commit_fetch) -> tuple[list, str]:
+    """
+    Retrieves relevant files modified between commits and the current repository head.
+
+    Args:
+        client (Client): An instance of the client used for interacting with the repository.
+        last_commit_fetch (str): The SHA of the last fetched commit.
+
+    Returns:
+        tuple: A tuple containing a list of relevant file URLs and the SHA of the current repository head.
+    """
+    current_repo_head = demisto.params().get('branch_head')
+    base_repo_sha = last_commit_fetch or client.get_repo_base_sha()
+    try:
+        all_commits_files = client.get_files_between_commits(base_repo_sha, current_repo_head) # type: ignore
+        relevant_files = parsing_files_by_status(all_commits_files)
+        return relevant_files, current_repo_head # type: ignore
```

Comment:
we're just returning `current_repo_head` that we got as input. How do we update to HEAD commit?